fractions/javaee/jaxrs/src/main/java/org/wildfly/swarm/jaxrs/runtime/DefaultApplicationDeploymentProcessor.java

```diff
@@ -76,7 +76,9 @@ public void process() throws Exception {
             return;
         }
 
-        addGeneratedApplication(archive.as(JAXRSArchive.class));
+        if (applicationPath.isExplicit()) {
+            addGeneratedApplication(archive.as(JAXRSArchive.class));
+        }
     }
```

Comment:
This is the only change that was required. All the other changes pertain to the test cases.